enhancements/storage/storage-security-policy.md

```diff
@@ -0,0 +1,198 @@
+
+We propose to add a new admission hook (which will be eventually replaced with MAP) which will trigger the defautling logic, similar to how SCC matching hooks are triggered.
+
+### Eventual replacement with Mutating Admission policy
+
+We want to replace admission plugin with Mutating Admission Policy once the feature becomes generally available. Once Mutating Admission Policy becomes GA, we propose to replace the admission hook with MAP on the lines of https://docs.google.com/document/d/13xa7imyvQzok3IJ66mheGqemQGhkNySl-He1h1iRuSM/edit?usp=sharing
+
+This would serve similar purpose as admission hook we are carrying as go code.
+
+### Proof of concept
+
+We have already implemented a proof-of-concept https://github.com/openshift/kubernetes/pull/2311/files, which works and does the job. But this uses existing SCC admission hook.
+
+
+### Workflow Description
+
+Admins that want to set defaults of `fsGroupChangePolicy` and `selinuxChangePolicy` will set `storage.openshift.io/fsgroup-change-policy` or `storage.openshift.io/selinux-change-policy`
+labels in corresponding namespaces.
+
+This will result in setting these fields in all pods in affected namespace if these fields are *not* already set.
+
+### API Extensions
+
+Adds following labels as API conventions for defaulting of security policies:
+
+- `storage.openshift.io/fsgroup-change-policy`
+- `storage.openshift.io/selinux-change-policy`
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+It should not have effect on Hypershift deployments.
+
+#### Standalone Clusters
+
+It should not have effect on Standalone deployments.
+
+#### Single-node Deployments or MicroShift
+
+It should not have effect on Single-Node deployments.
+
+### Implementation Details/Notes/Constraints
+
+We will implement the necessary admission hooks in - https://github.com/openshift/apiserver-library-go/tree/master/pkg ,
+which will be feature-gated while in Tech Preview.
+
+### Risks and Mitigations
+
+This feature requires storage to be enabled for this to work. See Operational Aspect section for more details.
+
+### Drawbacks
+
+The admission hook implementation should be replaced by Mutating Admission Hook and hence it is kinda busy work while we wait for MAP implementation to go GA in upstream.
+
+## Alternatives (Not Implemented)
+
+None
+
+## Open Questions [optional]
+
+None
+
+## Test Plan
+
+We will add e2e tests that validate if labels are working as intended. These would be either part of CSO tests or `openshift/origin` tests.
+
+## Graduation Criteria
+
+The feature will be behind a feature-gate during tech-preview.
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+During the initial development this feature will be behind a Openshift feature gate and will not be generally available. When we move this feature-gate to GA,
+it should be as simple as removal of the Openshift feature-gate.
```

Comment:
What testing will you put in place to make sure you are confident in the features ability to be promoted?

Reply:
This was already mentioned in Test Plan section btw. Let us know if I missed something.